webapp/channels/src/components/advanced_text_editor/send_button/send_post_options/core_menu_options.tsx

```diff
@@ -32,6 +37,36 @@ function CoreMenuOptions({handleOnSelect}: Props) {
         />
     );
 
+    const extraProps: Partial<MenuItemProps> = {};
+
+    if (isDM) {
+        function getScheduledTimeInTeammateTimezone(userCurrentTimestamp: number, teammateTimezoneString: string): string {
+            const scheduledTimeUTC = DateTime.fromMillis(userCurrentTimestamp, {zone: 'utc'});
+            const teammateScheduledTime = scheduledTimeUTC.setZone(teammateTimezoneString);
+            const formattedTime = teammateScheduledTime.toFormat('h:mm a');
+            return formattedTime;
+        }
+
+        const teammateTimezoneString = teammateTimezone.useAutomaticTimezone ? teammateTimezone.automaticTimezone : teammateTimezone.manualTimezone || 'UTC';
+
+        const dmTeammateTimezone = (
```

Comment:
Can you move this into a useMemo so thats its not re-computed on each render?

Reply:
the gain of using useMemo over this component is insignificant, however, this is render conditionally, that would mean for using the hook I would need to take it out of the condition and will be always computed on every render and will check memory and deps verifying if it should update.